src/rpc/blockchain.cpp

```diff
@@ -210,7 +210,7 @@ UniValue waitfornewblock(const JSONRPCRequest& request)
             + HelpExampleRpc("waitfornewblock", "1000")
         );
     int timeout = 0;
-    if (request.params.size() > 0)
+    if (request.params.size() > 0 && !request.params[0].isNull())
```

Comment:
In commit "check for null values in rpc args and handle appropriately"

Instead of writing `request.params.size() > X && !request.params[X].isNull()` everywhere, I think you should just prefer  `!request.params[X].isNull()`. The UniValue array implementation has bounds checking built in and already returns null for missing elements.